uvcombine/uvcombine.py

```diff
@@ -1002,39 +1075,110 @@ def feather_simple_cube(cube_hi, cube_lo,
+        else:
+            cube_lo_reproj = cube_lo
+
+        # Check that the pixel sizes of both cubes now match
+        equal_sizes = cube_lo_reproj.shape == cube_hi.shape
+        if not equal_sizes:
+            raise ValueError("The cube_lo array shape does not match the cube_hi"
+                             " shape. Enable `allow_lo_reproj` or reproject cube_lo"
+                             " before feathering.")
+
+        # Ensure spatial chunk sizes are matched.
+        if force_spatial_rechunk:
+            cube_hi = cube_hi.rechunk(('auto', -1, -1), **save_kwargs)
+            cube_lo = cube_lo.rechunk(('auto', -1, -1), **save_kwargs)
+
+            if cube_hi._data.chunksize != cube_lo._data.chunksize:
+                raise ValueError("The chunk size does not match between the cubes."
+                                 f" cube_hi: {cube_hi._data.chunksize} "
+                                 f" cube_lo: {cube_lo._data.chunksize} "
+                                 "Check reprojection or apply prior to feathering.")
+
+        # Check kwargs for feather_simple kwarg to allow matching units
+        match_units = kwargs.pop('match_units', True)
+
+        # Check for units consistency
+        if match_units:
+            # After this step, the units of cube_hi are some sort of surface brightness
+            # unit equivalent to that specified in the high-resolution header's units
+
+            # NOTE: is this memory-friendly??
+            cube_lo = cube_lo.to(cube_hi.unit)
+
+            # When in a per-beam unit, we need to scale the low res to the
+            # Jy / beam for the HIRES beam.
+            jybm_unit = u.Jy / u.beam
+            if cube_hi.unit.is_equivalent(jybm_unit):
+                cube_lo *= (cube_hi.beam.sr / cube_lo.beam.sr).decompose().value
+
+        # Add check that the units are compatible
+        equiv_units = cube_lo.unit.is_equivalent(cube_hi.unit)
+        if not equiv_units:
+            raise ValueError("Brightness units are not equivalent: "
+                            f"hires: {cube_hi.unit}; lowres: {cube_lo.unit}")
+
+        feathcube = _dask_feather_cubes(cube_hi, cube_lo,
+                                        save_to_tmp_dir=use_save_to_tmp_dir,
+                                        **kwargs)
 
-    pb = ProgressBar(cube_hi.shape[0])
-    for ii in range(cube_hi.shape[0]):
+    else:
 
-        hslc = cube_hi[ii]
-        lslc = cube_lo[ii]
+        if use_memmap:
+            from tempfile import NamedTemporaryFile
+            fname = NamedTemporaryFile()
+            feath_array = np.memmap(fname, shape=cube_hi.shape, dtype=float, mode='w+')
+        else:
+            feath_array = np.empty(cube_hi.shape)
 
-        feath_array[ii] = feather_simple(hslc, lslc, **kwargs)
+        pb = ProgressBar(cube_hi.shape[0])
+        for ii in range(cube_hi.shape[0]):
 
-        pb.update()
+            hslc = cube_hi[ii]
+            lslc = cube_lo[ii]
```

Comment:
I think here too?  I may have lost which function we're in though
```suggestion
            lslc = cube_lo_reproj[ii]
```

Reply:
Only needed fixing for the dask version. Otherwise `feather_simple` handles this per plane.